content/6_Advanced/DP_More.mdx

```diff
@@ -21,6 +19,20 @@ frequency: 1
 	<Resource source="GCP" title="15.4.3 - Knuth's Optimization"/>
 </Resources>
 
+Knuth's optimization is, in a way, a special case of Range DP. In general, it is used to solve DP problems with the following transition - $\texttt{dp}[i][j]$ = $\texttt{cost}[i][j] + min_{i≤k<j}(\texttt{dp}[i][k] + \texttt{dp}[k+1][j])$.  
+Further, the cost function must be monotonic on lattice intervals and it must satisfy the quadrangle inequality, i.e., for all $a\leq b\leq c \leq d$:  
+1. $cost[b][c] ≤ cost[a][d]$ 
+2. $cost[a][c] + cost[b][d] ≤ cost[a][d] + cost[b][c]$
+
+The optimization is as follows - define $\texttt{opt}[i][j] = argmin_{i≤k<j}(dp[i][k] + dp[k+1][j])$. In other words, $\texttt{opt}[i][j]$ is the index $k$ for which $\texttt{dp}[i][k] + \texttt{dp}[k+1][j]$ takes on its minimum value. If more than one such index exists, then take the minimum (or the maximum, doesn't matter). Then, assuming conditions on the dp transition and the cost function to be satisfied, we can show the following inequality - $\texttt{opt}[i][j-1] ≤ \texttt{opt}[i][j] ≤ \texttt{opt}[i+1][j]$.
```

Comment:
> we can show

don't see you proving this anywhere
best not to leave exercises to the reader

Reply:
The proof is a bit long. I'll add a link to it in the resources.

Reply:
is that the GFG link

Reply:
if it is can you say that it's in that link above

Reply:
Will do

Reply:
The commit history seems a bit messy, so just to clarify, I squashed the previous 5 commits (the updates you requested) as they were all one liners to a single commit (this should reflect on the log). I also added a mention in the article that the proof is in the resources.